### PR TITLE
Adds option to customize antd variables

### PR DESCRIPTION
**Pull request by lgenzelis**
This PR allows `antd` users to provide a file to override whichever `antd` variable they want.  I added instructions to use it to the `README` file.

Inspired by another plugin, [next-plugin-antd-less]

[next-plugin-antd-less]: https://github.com/SolidZORO/next-plugin-antd-less

### CODE REVIEW

**Comment by elado** [comment]
Hi @lgenzelis, thanks for contributing!
My intention was to keep this plugin as "dumb" as possible. I'm currently not using `additionalData` at all, but also haven't worked too much with Less.

So a couple of questions - 
- How common is it to add overrides with `additionalData`? Are there any other ways to do that?
- Would it be ok to just add `lessLoaderOptions` and spread it in the `options`, so users can add stuff in `next.config.js`? I'm asking this because users _may_ want to do additional things in `additionalData` and I wouldn't want to limit that to only overrides file load.

**Comment by lgenzelis** [comment]
You're welcome @elado ! ^^ I don't know how common is it to add overrides to `additionalData`, but I do know it is extremely common to override some of `antd` variables. It's the [official way](https://ant.design/docs/react/customize-theme) to do customize it. 

Adding something like `lessLoaderOptions` would be ok, I guess. Less code here in the plugin, but more code for the users. It's your call :) We can still add an example of how we'd use `lessLoaderOptions` to customize `antd` in the `README`, since many users are probably going to want to do that.

**Comment by lgenzelis** [comment]
We could name that option `additionalData` or `lessLoaderAdditionalData` (I think `lessLoaderOptions` would be confusing, it's more similar to `lessOptions` than to `additionalData`).

Then, user usage for this case would be something like:

```js
module.exports = withLess({
  future: {
    webpack5: true,
  },
  //....
  lessLoaderAdditionalData: content =>`${content}\n\n@import '${pathToLessFileWithVariables}';`,
}
```

Doesn't seem so terrible :)

**Comment by dryleaf** [comment]
I was wondering whether I could inject antd styles without using `_app.js` page.

**Comment by jayliew** [comment]
> You're welcome @elado ! ^^ I don't know how common is it to add overrides to `additionalData`, but I do know it is extremely common to override some of `antd` variables. It's the [official way](https://ant.design/docs/react/customize-theme) to do customize it.

I just wanted to also chime in that I am also looking for a way to override some of the variables (in order to change color). Thank you all for working on this!

**Comment by lgenzelis** [comment]
I closed this PR and created a new one, #4, since we ended up going in a different direction that I originally described. You're absolutely right @elado , this is a better approach, it adds almost no complexity to the plugin, while still allowing users to customize as needed. 

BTW, the reason why I prefer overriding the variables using `additionalData` instead of just `modifyVars`, is that with the latter you have to stop the server and do `npm run dev` again each time you modify a variable. Using `additionalData` to inject the variables file, hot reload works as usual. It makes tweaking the variables much easier :)

File: README.md [hunk 1]
```diff
@@ -61,3 +61,30 @@ module.exports = withPlugins(plugins, {
   },
 });
 ```
+
+### Customize `antd` theme
+
+To override some of `antd` [default variables](https://ant.design/docs/react/customize-theme), just add them under `lessLoaderOptions.lessOptions.modifyVars`:
```
[the remark this answers is not in the document]
Comment (lgenzelis) [reply README.md]: Ok! I'll do that in the new PR, #4

**Comment by lgenzelis** [comment]
> I was wondering whether I could inject antd styles without using `_app.js` page. The [example](https://www.github.com/elado/next-with-less/tree/main/examples%2Fnext-with-less-example%2Fpages%2F_app.js) in this package relies merely on additional customization to NextJs's default `_app.js`.

@dryleaf I don't think that's possible. You need to import the styles somewhere, and `_app` is the place for that (since these are global styles).

**Comment by dryleaf** [comment]
@lgenzelis I created an [issue#3](https://github.com/elado/next-with-less/issues/3), please have a look.

**Comment by lgenzelis** [comment]
@dryleaf yeah, I get that warning as well, but Idk if there's anyway around it. Just in case, note that this is hurting build time performance. Site execution is not affected.

**Pull request closed without merging**